cluster/cluster.go

```diff
@@ -32,13 +33,34 @@ func Join(ctx context.Context, cfg Config) (*Cluster, error) {
 		zap.ReplaceGlobals(logger)
 	}
 
+	if cfg.etcdConfig.ClusterState == embed.ClusterStateFlagNew || len(clientUrls) == 0 {
+		clientUrls = urlsToString(cfg.etcdConfig.LCUrls)
+	} else if cfg.etcdConfig.ClusterState == embed.ClusterStateFlagExisting && len(clientUrls) == 0 {
+		return nil, fmt.Errorf("joining an existing cluster requires at least one client url from a member from the existing cluster")
+	}
+
+	clientCfg := clientv3.Config{
+		Endpoints:   clientUrls,
+		DialTimeout: 5 * time.Second,
+	}
+	client, err := clientv3.New(clientCfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create etcd client from config: %w", err)
+	}
+
+	if cfg.etcdConfig.ClusterState == embed.ClusterStateFlagExisting {
```

Comment:
Instead of this, I could just try to check if the cluster is existing if none of the peer urls in the config match any urls in the `InitialCluster` in the config. My concern with this is if the default config for `ClusterState` is `new`. Explicitly stating might be the better option unless there's something I didn't think of.